### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements for Dropdown Template

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1516]
## Description
This task describes the Storybook implementation for the Dropdown template as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for the Dropdown template can be found here: [Dropdown Notes](https://www.notion.so/matuzo/Dropdown-112f3f6e1bc080118c66eda5256a81f9).

---------------------------------

## Comment
**Problem:** When you tab into the content and press Esc, focus is lost. It's strange because it works in the example in [@sd-dropdown](https://www.notion.so/matuzo/sd-dropdown-112f3f6e1bc080dab9cae2a0771aa8e9). Edit from Refinement => seems like it because there is no content in the dropwdown. Probably doesn't work here as well.
**Solution:** Put focus back on the button.

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
